File: LAB02/task3.py
from file_io import read, write


def insertion_sort(a, b):
    for i in range(1, len(b)):
        j = i - 1
        cur_a = a[i]
        cur_b = b[i]
        while j >= 0 and cur_b > b[j]:
            a[j + 1] = a[j]
            a[j] = cur_a
            b[j + 1] = b[j]
            b[j] = cur_b
            j -= 1


a, b = read("task3-input1.txt")
insertion_sort(a, b)
write("task3-output1.txt", a)

c, d = read("task3-input2.txt")
insertion_sort(c, d)
write("task3-output2.txt", c)

File: LAB05/task3.cpp
#include <bits/stdc++.h>
using namespace std;

void execute(string input_file, string output_file) {
    freopen(input_file.c_str(), "r", stdin);
    freopen(output_file.c_str(), "w", stdout);

    int n, jack_time = 0, jill_time = 0;
    cin >> n;
    vector<int> times(n);
    for (auto &u : times) {
        cin >> u;
    }
    sort(times.begin(), times.end());
    string turn, sequence = "";
    cin >> turn;
    queue<int> q;
    stack<int> s;
    for (auto &t : times) {
        q.push(t);
    }
    for (int i = 0; i < turn.size(); i++) {
        if (turn[i] == 'J') {
            if (!q.empty()) {
                s.push(q.front());
                jack_time += q.front();
                sequence += to_string(q.front());
                q.pop();                
            }
        } else {
            if (!s.empty()) {
                jill_time += s.top();
                sequence += to_string(s.top());
                s.pop();
            }
        }
    }
    cout << sequence << endl;
    cout << "Jack will work for " << jack_time << " hours\nJill will work for " << jill_time << " hours";
}

int main() {
    execute("task3-input1.txt", "task3-output1.txt");
    execute("task3-input2.txt", "task3-output2.txt");
}